### PR TITLE
[7.59.x][JBPM-9799] - Create auto-generated PIM DDL scripts by Hibernate

### PR DESCRIPTION
**Pull request by afalhambra**
Backport of **[JBPM-9799](https://issues.redhat.com/browse/JBPM-9799)**: Create auto-generated PIM DDL scripts by Hibernate

### CODE REVIEW

**Comment by mareknovotny** [comment]
i guess @afalhambra you need to update also this PR to pass green ;)

**Comment by afalhambra** [comment]
Thanks @mareknovotny - will update the PR but I would like to wait for @gmunozfe review before merging.

**Comment by mareknovotny** [comment]
np

File: examples/flyway/flyway.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+quarkus:
+  flyway:
+      connect-retries: 10
+      table: flyway_pim_history
+      migrate-at-start: true
+      baseline-on-migrate: true
+      baseline-version: 1.0
+      baseline-description: PimDB
+      sql-migration-prefix: PIM
```
Comment (gmunozfe) [inline examples/flyway/flyway.yml:9]: Is this right? Shouldn't it refer to the database?

File: src/main/resources/application.yaml [hunk 2]
```diff
@@ -38,9 +38,18 @@ quarkus:
...
+    table: flyway_pim_history
+    migrate-at-start: true
+    baseline-on-migrate: true
+    baseline-version: 1.0
+    baseline-description: PimDB
+    sql-migration-prefix: h2
+  # Quartz configuration
   quartz:
-    store-type: ram
+    store-type: jdbc-cmt
```
Comment (gmunozfe) [inline src/main/resources/application.yaml:52]: Shouldn't it be jdbc-tx or ram? (according to README)

File: src/test/filtered-resources/application.yaml [hunk 3]
```diff
@@ -2,21 +2,30 @@ quarkus:
...
+    table: flyway_pim_history
+    migrate-at-start: true
+    baseline-version: 1.0
+    baseline-description: PimDB
+    sql-migration-prefix: ${maven.jdbc.db-kind}
+    clean-at-start: true
   # Quartz configuration
   quartz:
-    store-type: ram
+    store-type: jdbc-cmt
```
Comment (gmunozfe) [inline src/test/filtered-resources/application.yaml:16]: Same here

File: ddl-scripts/h2/h2-quartz-schema.sql [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+CREATE TABLE QRTZ_SIMPLE_TRIGGERS (
+  SCHED_NAME VARCHAR(120) NOT NULL,
+  TRIGGER_NAME VARCHAR (200)  NOT NULL ,
+  TRIGGER_GROUP VARCHAR (200)  NOT NULL ,
+  REPEAT_COUNT BIGINT NOT NULL ,
+  REPEAT_INTERVAL BIGINT NOT NULL ,
+  TIMES_TRIGGERED BIGINT NOT NULL
+);
+
+CREATE TABLE qrtz_simprop_triggers
```
Comment (gmunozfe) [inline ddl-scripts/h2/h2-quartz-schema.sql:80]: Is it lowercase for any reason?

File: pom.xml [hunk 5]
```diff
@@ -479,5 +569,97 @@
...
+      <id>mariadb</id>
+      <activation>
+        <property>
+          <name>maven.jdbc.driver.class</name>
+          <value>org.mariadb.jdbc.Driver</value>
+        </property>
+      </activation>
+      <properties>
+        <maven.jdbc.db-kind>mariadb</maven.jdbc.db-kind>
+        <maven.pim.flyway.location>db/migration/mariadb</maven.pim.flyway.location>
```
Comment (gmunozfe) [inline pom.xml:622]: This profile is the only one that contains "maven.pim.flyway.location", shouldn't it be the same for all?

File: src/test/java/org/kie/processmigration/persistence/DDLScriptsTest.java [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+        // Then
+        if (execution.getType() == ASYNC) {
+            assertThat(scheduler.checkExists(new JobKey(migration.getId().toString())), is(Boolean.TRUE));
+            assertThat(scheduler.checkExists(new TriggerKey(migration.getId().toString())), is(Boolean.TRUE));
+            if (!count.await(10, TimeUnit.SECONDS)) {
+                fail("Failed while waiting for the jobs to be completed");
+            }
+        }
+
+        // Then
```
Comment (gmunozfe) [inline src/test/java/org/kie/processmigration/persistence/DDLScriptsTest.java:164]: Misleading comment :confused:

File: README.md [hunk 7]
```diff
@@ -152,14 +152,23 @@ quarkus:
...
+    table: flyway_pim_history
+    migrate-at-start: true
+    baseline-on-migrate: true
+    baseline-version: 1.0
+    baseline-description: PimDB
+    sql-migration-prefix: h2 (1)        
 # Quartz configuration
   quartz:
-    store-type: ram
+    store-type: jdbc-cmt
```
Comment (gmunozfe) [inline README.md:166]: Shouldn't it be `ram` or `jdbc-tx` according to the paragraph below?

**Review by gmunozfe — APPROVED** [review]
Looks very good, top-notch work @afalhambra !

Some minor comments, you can go for them in the next iterations.

Also take into account this db2 issue fixed at Quartz: https://github.com/quartz-scheduler/quartz/pull/559

File: examples/flyway/flyway.yml (hunk 1 above)
Comment (afalhambra) [reply examples/flyway/flyway.yml:9]: you're right, it should be the database, not `PIM`. Added for the next iteration as this is in an example file.

File: src/main/resources/application.yaml (hunk 2 above)
Comment (afalhambra) [reply src/main/resources/application.yaml:52]: It should be `jdbc-cmt` - In the README, it is stating that the quartz is using a in-memory database managed by the container, which is `jdbc-cmt`. I will update the README file to clarify this. Thanks!

File: ddl-scripts/h2/h2-quartz-schema.sql (hunk 4 above)
Comment (afalhambra) [reply ddl-scripts/h2/h2-quartz-schema.sql:80]: no, not really, it's picked up from the quartz repo, I will re-format the scripts.

File: pom.xml (hunk 5 above)
Comment (afalhambra) [reply pom.xml:622]: Thanks for that! it's not needed. Added to the next iteration

File: src/test/java/org/kie/processmigration/persistence/DDLScriptsTest.java (hunk 6 above)
Comment (afalhambra) [reply src/test/java/org/kie/processmigration/persistence/DDLScriptsTest.java:164]: ok

File: README.md (hunk 7 above)
Comment (afalhambra) [reply README.md:166]: same

**Comment by afalhambra** [comment]
> Looks very good, top-notch work @afalhambra !
> 
> Some minor comments, you can go for them in the next iterations.
> 
> Also take into account this db2 issue fixed at Quartz: [quartz-scheduler/quartz#559](https://github.com/quartz-scheduler/quartz/pull/559)

Thanks a lot @gmunozfe - as they're minor issues I filed a new jira ticket to track them down https://issues.redhat.com/browse/JBPM-9889 and incorporate them in the next iteration.

@mareknovotny Please feel free to merge. Thanks!

**Comment by mareknovotny** [comment]
i agree as i already merged PR to main and these post review changes need to be done for both branches

**Pull request merged**